test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -361,6 +392,23 @@ public void ParseAssignment ()
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE='"));
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE=0\n1"));
 
+            testParse("EV_DNE", "", "EV_DNE=");
+            testParse("EV_DNE", "a b c", "EV_DNE=a b c # comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a'bc allow singleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a\"bc allow doubleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test #a$bc allow dollarSign in comment");
+
+            testParse("EV_DNE", "http://www.google.com/#anchor", "EV_DNE=http://www.google.com/#anchor");
+            testParse("EV_DNE", "", "EV_DNE= #no value just comment");
+            testParse("EV_DNE", "", "EV_DNE=#no value just comment");
+            testParse("EV_DNE", "a'b''c", "EV_DNE=a'b''c #allow inline singleQuotes in unquoted values");
+            testParse("EV_DNE", "a\"b\"\"c", "EV_DNE=a\"b\"\"c #allow inline doubleQuotes in unquoted values");
+
+            testParse("EV_DNE", "a'b''c", "EV_DNE=\"a'b''c\" #this already works");
+            testParse("EV_DNE", "a\"b\"\"c", "EV_DNE='a\"b\"\"c' #this already works");
+            testParse("EV_DNE", "a 'b' c", "EV_DNE=\"a 'b' c\" #this already works");
```

Comment:
In this version, fairly intentionally, there are no quoted values in the middle of an unquoted value (i.e. removing the outer quotes would make this test fail). By allowing quotes in unquoted values, we are starting to get into weird territory where `a'b'c` works but `a 'b' c` does not, and this is part of why I am uncomfortable with allowing even `a'b'c` to begin with. The quoted values handle these cases just fine already.

Reply:
Because of this weirdness, I removed support for quote chars inside unquoted values entirely -- see the tests on this, bash + fish do not handle these the way unquoted values would have (in keeping the quote chars as they are)